framework/log/DbTarget.php

```diff
@@ -60,6 +60,12 @@ public function init()
      */
     public function export()
     {
+        if ($this->db->getTransaction()) {
+            $this->db = clone $this->db;
+            $this->db->pdo = null;
```

Comment:
master-slave should work well. I see no reason for it not to.

Transaction property — yes. Should be reset. Good catch.